include/revng/BasicAnalyses/GeneratedCodeBasicInfo.h

```diff
@@ -256,6 +256,10 @@ class GeneratedCodeBasicInfo {
     return isJump(BB->getTerminator());
   }
 
+  void removeJTBlockFromDT(llvm::BasicBlock *BB) {
+    purgeDomTree(BB->getParent());
+  }
```

Comment:
1. `removeJTBlockFromDT` is exclusively used after calling `GCBI->getJumpTarget(BB);`, thus BB is always a JT. No meant to be used anywhere else.
2. `removeJTBlockFromDomTree` sounds better?
3. Not close in term of code? `purgeDomTree` belongs to `class CSVsUsage` which is a nested class of GCBI. Can I directly expose `purgeDomTree` without using `removeJTBlockFromDomTree`?
4. Keep getting crashes if I do not erase the element from `DTMap`.

Reply:
1. It doesn't matter. This is not a private function, it might be misused.

"Keep getting crashes" is not a good explanation. You need to have a clear idea of what's going on here.

Why are you removing a jump target? It doesn't seem like something SA2 should do.
Is it a cloned basic block?

Reply:
Now it should look fairly good.